src/app/content/components/NudgeStudyTools/index.tsx

```diff
@@ -0,0 +1,131 @@
+  const document = assertDocument();
+  const wrapperRef = React.useRef<HTMLElement>(null);
+  const isMobile = useMatchMobileQuery();
+  const positions = usePositions(isMobile);
+  const dispatch = useDispatch();
+
+  React.useEffect(() => {
+    if (positions) {
+      document.body.style.overflow = 'hidden';
+    }
+    return () => { document.body.style.overflow = null; };
+    // document will not change
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [positions]);
+
+  React.useEffect(() => {
+    const element = wrapperRef.current;
+    if (positions && element) {
+     element.focus();
+    }
+  }, [wrapperRef, positions]);
+
+  if (!positions) { return null; }
+
+  return ReactDOM.createPortal(
+    <NudgeWrapper data-analytics-region='Nudge Study Tools'>
+      <NudgeArrow
+        src={isMobile ? arrowMobile : arrowDesktop}
+        alt=''
+        top={positions.arrowTopOffset}
+        left={positions.arrowLeft}
+      />
+      <NudgeCloseButton
+        tabIndex={2}
+        top={positions.closeButtonTopOffset}
+        left={positions.closeButtonLeft}
+        onClick={() => dispatch(closeNudgeStudyTools())}
+        data-analytics-label='close'
+      >
+        <NudgeCloseIcon />
+      </NudgeCloseButton>
+      <FormattedMessage id='i18n:nudge:study-tools:aria-label'>
+        {(msg: string) => <NudgeContentWrapper
+          ref={wrapperRef}
+          tabIndex={1}
+          aria-label={msg}
+          top={positions.contentWrapperTopOffset}
+          right={positions.contentWrapperRight}
+        >
+          <NudgeContent>
+            <NudgeHeading />
+            <NudgeText />
+          </NudgeContent>
+        </NudgeContentWrapper>}
+      </FormattedMessage>
+      <NudgeBackground>
+        <NudgeSpotlight
+          top={positions.spotlightTopOffset}
+          left={positions.spotlightLeftOffset}
+          height={positions.spotlightHeight}
+          width={positions.spotlightWidth}
+        />
+      </NudgeBackground>
+    </NudgeWrapper>,
+    document.body
+  );
+};
+
+// Do not render <NudgeStudyTools/> if it is hidden so scroll listener is not attached
+// to the DOM and do not render if document or window is undefined which may happen for prerendering.
+// tslint:disable-next-line: variable-name
+const NoopForPrerenderingAndForHiddenState = () => {
+  const hasStudyGuides = useSelector(hasStudyGuidesSelector);
+  const show = useSelector(showNudgeStudyTools);
+  const trackOpen = useAnalyticsEvent('openNudgeStudyTools');
+  const dispatch = useDispatch();
+
+  useIncrementPageCounter();
+
+  React.useEffect(() => {
+    if (
+      show === null
+      && hasStudyGuides
+      && shouldDisplayNudgeStudyTools()
+    ) {
+      setNudgeStudyToolsCookies();
+      trackOpen();
+      dispatch(openNudgeStudyTools());
+    }
+  }, [show, hasStudyGuides, trackOpen, dispatch]);
+
+  if (typeof document === 'undefined' || typeof window === 'undefined' || !show ) { return null; }
```

Comment:
its sometimes important for the initial client render to match the server render exactly, so instead of doing this check in the render, you can do a didMount hook that sets a state param that controls when this is rendered.

or you could just use `assertDocument` less aggressively in `NudgeStudyTools`

Reply:
pretty sure you could just delete the document and window stuff from this condition, because the useEffect will not trigger the dispatch in prerender